src/meter/Meter.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import { props, t, skinnable } from '../utils';
+import cx from 'classnames';
+import FlexView from '../flex/FlexView';
+import find from 'lodash/collection/find';
+import every from 'lodash/collection/every';
+import uniq from 'lodash/array/uniq';
+
+const Range = t.refinement(t.struct({
+  startValue: t.Number,
+  endValue: t.Number,
+  fillingColor: t.maybe(t.String),
+  labelColor: t.maybe(t.String)
+}), r => r.startValue < r.endValue, 'Range');
+
+const Ranges = t.refinement(t.list(Range), (rangeList) => {
+
+  const stepsWithout = (steps, stepToRemove) => {
+    const stepsNoDuplicate = uniq(steps);
+    return stepsNoDuplicate.filter( step => (
+      !(step.startValue === stepToRemove.startValue && step.endValue === stepToRemove.endValue)
+    ));
+  };
+
+  const isOverlapped = (step1, step2) => (
+    Math.max(step1.startValue, step2.startValue) < Math.min(step1.endValue, step2.endValue)
+  );
+
+  const noOverlappingRanges = (steps) => {
+    return every((steps), step1 => {
+      return every(stepsWithout(steps, step1), step2 => {
+        return !isOverlapped(step1, step2);
+      });
+    });
+  };
```

Comment:
Yes definitively it is. That props validation is actually a bit complex, and there no easy way of doing that. To check if a range is overlapped i have to compare each range with every other range except himself (all of this because rangeList is not sorted). If you have any refactoring proposal are  appreciated.